lib/python/picongpu/picmi/simulation.py

```diff
@@ -299,13 +293,12 @@ def __get_init_manager(self) -> species.InitManager:
         3. generate operations which have inter-species dependencies
         4. generate operations without inter-species dependencies
         """
-        initmgr = species.InitManager()
+        initmgr = pypicongpu.species.InitManager()
 
-        # check preconditions
+        # check preconditions, @todo move to picmistandard, Brian Marre 2024
         assert len(self.species) == len(self.layouts)
 
-        # either no layout AND no profile, or both
-        # (also: no ratio without layout and profile)
+        # check either no layout AND no profile, or both and ratio only set if leyout and profile also set
```

Comment:
```suggestion
        # check either no layout AND no profile, or both and ratio only set if layout and profile also set
```
But I'd say that this is redundant. It is exactly what the code does (and even explicitly tells you in the assertion).

Reply:
moved to separate function as requested.

Reply:
comment removed